Integrations/integration-ServiceNowPython.yml

```diff
@@ -0,0 +1,1728 @@
+            'Contents': res,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Link added to ServiceNow ticket\n', hr, removeNull = True)
+        }
+
+        return entry
+
+
+    def add_link(ticket_id, ticket_type, key, link):
+        body = {}
+        body[key] = link
+        path = 'table/' + ticket_type + '/' + ticket_id
+
+        return send_request(path, 'patch', body=body)
+
+    def add_comment_command():
+        ticket_id = demisto.args()['id']
+        key = 'comments' if demisto.args().get('post-as-comment', 'false') == 'true' else 'work_notes'
+        text = demisto.args()['comment']
+        ticket_type = get_table_name(demisto.args().get('ticket_type', None))
+
+        res = add_comment(ticket_id, ticket_type, key, text)
+
+        hr = get_ticket_human_readable(res['result'])
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': res,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Comment added to ServiceNow ticket\n', hr, removeNull = True)
+        }
+
+        return entry
+
+    def add_comment(ticket_id, ticket_type, key, text):
+        body = {}
+        body[key] = text
+        path = 'table/' + ticket_type + '/' + ticket_id
+
+        return send_request(path, 'patch', body=body)
+
+    def query_command():
+        sysparm_limit = demisto.args().get('limit', DEFAULTS['limit'])
+        sysparm_query = demisto.args().get('sysparm_query', None)
+        ticket_type = get_table_name(demisto.args().get('ticket_type', None))
+
+        res = query(ticket_type, sysparm_limit, sysparm_query)
+
+        if not res or 'result' not in res:
+            return 'Cannot find incident'
+
+        hr = get_ticket_human_readable(res['result'])
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': res,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('ServiceNow ticket', hr, removeNull = True),
+            'EntryContext': {
+                  'Ticket(val.ID==obj.ID)': get_ticket_context(res['result']),
+            }
+        }
+
+        return entry
+
+
+    def query(ticket_type, sysparm_limit, sysparm_query):
+        query_params = {}
+        query_params['sysparm_limit'] = sysparm_limit
+        if sysparm_query:
+            query_params['sysparm_query'] = sysparm_query
+
+        path = 'table/' + ticket_type
+
+        return send_request(path, 'get', params = query_params)
+
+
+    def upload_file_command():
+        ticket_type = get_table_name(demisto.args().get('ticket_type', None))
+        ticket_id = demisto.args()['id']
+        file_id = demisto.args()['file_id']
+        file_name = demisto.args().get('file_name',
```

Comment:
use `getFilePath` to get the name of the file: the function returns a dict with name, path as keys.

Reply:
The file can be given as an argument, not necessarily from context